### ISSUE TITLE
Use activationHooks to activate the package

### ISSUE THREAD

**Issue by fnkr** [AtomLinter/linter-php#86]
After the latest update when I start Atom I get the following error:

```
Failed to spawn command php. Make sure php is installed and on your PATH
```

Yes, PHP is not in my path _right now_, nevertheless the error should only come when I open (lint) a PHP file. I sync my Atom installation across multiple systems and PHP is only installed where I need it.

**Comment by Arcanemagus** [comment]
You want to support having the package installed, but be purposely in an unusable state?

**Comment by fnkr** [comment]
Yes. As long as you don't need the linter it should not produce any errors. That's how all the other linters I use behave. And this linter, at least until v1.1.0.

**Comment by steelbrain** [comment]
@fnkr but at least the machine should have PHP installed. I don't see anything wrong with it. This package needs PHP installed to check the version of phpcs installed

**Comment by Arcanemagus** [comment]
@steelbrain this is `linter-php` not `linter-phpcs`.

@fnkr What is the case for having this package installed if you aren't going to use it? What case would there be that simply disabling this package or not installing it in the first place wouldn't also solve?

**Comment by fnkr** [comment]
@Arcanemagus As I said I sync my Atom config, including packages, across different systems/VMs. I do not develop PHP on each system so I don't want to install it on each system. I dont want to install Python/PHP/JavaScript/node.js/Golang/Ruby/Perl/C++/C on all of my systems and until now none of the linters I have installed errored as long as I don't try to edit a file in a programming language I have not installed on the system.

**Comment by Arcanemagus** [comment]
@fnkr the `linter-*` packages currently as a whole have terrible onboarding for new users, this was my first bit of attempting to help with new users to make sure they have things set up properly.

Looking into this though has brought `activationHooks` to my attention, which should make the package only activate when `language-php` is first used. So unless you open a PHP file the package shouldn't activate. I'll put up a PR for that shortly.

**Comment by fnkr** [comment]
@Arcanemagus Huge thanks for that!

**Comment by Arcanemagus** [comment]
Finally released this as v1.1.1, getting the specs working took a bit of work with @steelbrain :wink:.

**Comment by Arcanemagus** [comment]
So `activationHooks` was removed only temporarily, v1.4.0 supposedly fixed the major activation issues that caused me to remove it... but I have been unable to get the specs working on v1.4.0 when I have `activationHooks` setup :confused:.

As soon as I can get the specs working, I'll be pushing out a version that supports that!

**Comment by Arcanemagus** [comment]
For now you can simply force the installation of v1.1.6 to get back to that behavior, nothing big has been updated since then.

**Comment by joseluisch** [comment]
because  this error
![index php cxampphtdocsmi pagina atom](https://cloud.githubusercontent.com/assets/28614760/25982319/88466328-36a9-11e7-8511-b657c2cd1b40.jpg)
![index php mi pagina cxampphtdocsmi pagina atom](https://cloud.githubusercontent.com/assets/28614760/25982320/8849d454-36a9-11e7-9fee-79b2abe99ad5.jpg)

**Comment by Arcanemagus** [comment]
@joseluisch This issue has absolutely nothing to do with the error you are showing there. Please read the message in the console and if you are still having problems please file a **new** issue so we can help work you through this.
